File: models/PastMedicalHistory.js
import mongoose from "mongoose";

const PastMedicalHistorySchema = new mongoose.Schema({
    patientId: { type: mongoose.Schema.Types.ObjectId, ref: "PatientRegistration" },
    disease: String,
    duration: String,
    medication: String,
    createdByUser:{type:mongoose.Schema.Types.ObjectId,ref:"User"},
  }, { timestamps: true });
  
  const PastMedicalHistory= mongoose.model("PastMedicalHistory", PastMedicalHistorySchema);
  export default PastMedicalHistory;
  

File: jstojv/src/main/java/com/samrat/jstojv/entity/PastMedicalHistory.java
package com.samrat.jstojv.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "past_medical_histories")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PastMedicalHistory extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "patient_id")
    private PatientRegistration patient;

    private String disease;
    private String duration;
    private String medication;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by_user_id")
    private User createdByUser;
}
